src/main/java/Duke.java

```diff
@@ -1,61 +1,159 @@
-        System.out.println("    ____________________________________________________________");
+        taskCountMessage(itemCount);
+        printDivider();
     }
 
-    public static void main(String[] args) {
-        System.out.println("____________________________________________________________");
-        System.out.println(" Hello! I'm Duke");
-        System.out.println(" What can I do for you?");
-        System.out.println("____________________________________________________________");
+    private void addDeadline(String message) {
+        String[] removeActionMessage = message.split(" ", 2);
+        String[] params = removeActionMessage[1].split("/", 2);
+        tasks[itemCount] = new Deadline(params[0], params[1]);
+        echoTask(itemCount);
+        itemCount++;
+        taskCountMessage(itemCount);
+        printDivider();
+    }
 
+    private void addEvent(String message) {
+        String[] removeActionMessage = message.split(" ", 2);
+        String[] params = removeActionMessage[1].split("/", 2);
+        tasks[itemCount] = new Event(params[0], params[1]);
+        echoTask(itemCount);
+        itemCount++;
+        taskCountMessage(itemCount);
+        printDivider();
+    }
+
+    private void markTaskAsDone(String message) {
+        String[] params = message.split(" ");
+        int index = Integer.parseInt(params[1]) - 1;
+        tasks[index].setDone();
+        markAsDoneMessage(index);
+        printDivider();
+    }
+
+    private void invalidUserInputMessage() {
+        //TODO improve on invalid input message
+        System.out.println("wrong command");
+        printDivider();
+    }
+
+    public void run() {
+        greet();
         //initialise variables for scanning
-        String line = "";
-        Scanner in = new Scanner(System.in);
+        String userInput;
+        Scanner scanner = new Scanner(System.in);
 
         //while user input != bye, case insensitive
-        while (!line.equalsIgnoreCase("bye")) {
-            line = in.nextLine();
-            Echo(line);
+        while (true) {
+            userInput = scanner.nextLine();
+
+            if (userInput.equalsIgnoreCase(COMMAND_EXIT)) {
+                exit();
+                break;
+            }
+
+            if (userInput.equalsIgnoreCase(COMMAND_LIST_TASKS)) {
+                listTasks();
+                continue;
+            }
+
+            if (userInput.startsWith(COMMAND_ADD_TODO)) {
+                addTodo(userInput);
+                continue;
+            }
+
+            if (userInput.startsWith(COMMAND_ADD_DEADLINE)) {
+                addDeadline(userInput);
+                continue;
+            }
+            if (userInput.startsWith(COMMAND_ADD_EVENT)) {
+                addEvent(userInput);
+                continue;
+            }
+
+            if (userInput.startsWith(COMMAND_MARK_TASK_AS_DONE)) {
+                markTaskAsDone(userInput);
+                continue;
+            }
```

Comment:
Might be better to put the statements together . It'll help to reduce your LOC for this method.